### PR TITLE
Add validation for concentration intervals

### PR DESCRIPTION
**Pull request by islean**
## Description

The concentration needs to be validated that it is within the allowed bounds for the application, if the customer wants to skip reception control.

### Added

- Validation of the concentration fallen in the allowed interval when skipping reception control.

### Changed

-

### Fixed

-

### How to prepare for test

- [ ] Ssh to relevant server (depending on type of change)
- [ ] Use stage: `us`
- [ ] Paxa the environment: `paxa`
- [ ] Install on stage (example for Hasta):
    ```shell
    bash /home/proj/production/servers/resources/hasta.scilifelab.se/update-tool-stage.sh -e S_cg -t cg -b [THIS-BRANCH-NAME] -a
    ```

### How to test

- [ ] Do ...

### Expected test outcome

- [ ] Check that ...
- [ ] Take a screenshot and attach or copy/paste the output.

## Review

- [ ] Tests executed by
- [ ] "Merge and deploy" approved by
Thanks for filling in who performed the code review and the test!

### This [version](https://semver.org/) is a

- [ ] **MAJOR** - when you make incompatible API changes
- [ ] **MINOR** - when you add functionality in a backwards compatible manner
- [ ] **PATCH** - when you make backwards compatible bug fixes or documentation/instructions

## Implementation Plan

- [ ] Document in ...
- [ ] Deploy this branch on ...
- [ ] Inform to ...

### CODE REVIEW

File: cg/services/order_validation_service/workflows/tomte/validation/inter_field/utils.py [hunk 1]
```diff
@@ -158,3 +162,53 @@ def validate_subject_ids_in_case(case: TomteCase) -> list[SubjectIdSameAsCaseNam
...
+def is_sample_concentration_allowed(sample: TomteSample, application: Application):
+    concentration = sample.concentration_ng_ul
+    is_cfdna = is_sample_cfdna(sample)
+    interval = get_concentration_interval(application=application, is_cfdna=is_cfdna)
+    return is_sample_concentration_within_interval(concentration=concentration, interval=interval)
+
+
+def is_sample_cfdna(sample: TomteSample):
+    source = sample.source
+    return source == SourceType.CELL_FREE_DNA
```
Comment (Vince-janv) [inline cg/services/order_validation_service/workflows/tomte/validation/inter_field/utils.py:201]: oof, I hate that we have to implement this logic 😢  I wonder if it is even being used. No comment on the implementation though. Just wanted to ventilate

File: cg/services/order_validation_service/models/errors.py [hunk 2]
```diff
@@ -155,6 +155,17 @@ class SubjectIdSameAsSampleNameError(CaseSampleError):
     message: str = "Subject id must be different from the sample name"
 
 
+class InvalidConcentrationIfSkipRCError(CaseSampleError):
+    def __init__(self, case_name: str, sample_name: str, allowed_interval: tuple[int, int]):
+        field: str = "concentration_ng_ul"
+        message: str = (
+            f"Concentration must be between {allowed_interval[0]} ng/μL and {allowed_interval[1]} ng/μL"
```
Comment (Vince-janv) [inline cg/services/order_validation_service/models/errors.py:162]: Probably relevant to add this 
```suggestion
            f"Concentration must be between {allowed_interval[0]} ng/μL and {allowed_interval[1]} ng/μL if reception control should be skipped"
```

**Review by Vince-janv — APPROVED** [review]
🚀

File: cg/services/order_validation_service/models/errors.py (hunk 2 above)
Comment (islean) [reply cg/services/order_validation_service/models/errors.py:162]: Nice catch!

File: cg/services/order_validation_service/workflows/tomte/validation/inter_field/utils.py (hunk 1 above)
Comment (islean) [reply cg/services/order_validation_service/workflows/tomte/validation/inter_field/utils.py:201]: It was a quite recent addition with the skip rc flow, so I would guess that there has been use cases for it...

**Pull request merged**